src/main/java/org/wso2/carbon/connector/operations/EmailSend.java

```diff
@@ -0,0 +1,115 @@
+            ResponseHandler.setErrorsInMessage(messageContext, Error.CONNECTIVITY);
+            handleException(e.getMessage(), e, messageContext);
+        } catch (InvalidConfigurationException e) {
+            ResponseHandler.setErrorsInMessage(messageContext, Error.INVALID_CONFIGURATION);
+            handleException(e.getMessage(), e, messageContext);
+        } catch (ContentBuilderException e) {
+            ResponseHandler.setErrorsInMessage(messageContext, Error.RESPONSE_GENERATION);
+            handleException(e.getMessage(), e, messageContext);
+        }
+    }
+
+    /**
+     * Sends an email
+     *
+     * @param messageContext The message context that is generated for sending the email
+     * @param session        Mail Session to be used
+     * @return a result status indicating whether the email send is successful or not
+     */
+    private void sendMessage(MessageContext messageContext, EmailConnection session) throws EmailConnectionException,
+            InvalidConfigurationException {
+
+        String to = (String) getParameter(messageContext, EmailConstants.TO);
+        String from = (String) getParameter(messageContext, EmailConstants.FROM);
+        String cc = (String) getParameter(messageContext, EmailConstants.CC);
+        String bcc = (String) getParameter(messageContext, EmailConstants.BCC);
+        String replyTo = (String) getParameter(messageContext, EmailConstants.REPLY_TO);
+        String subject = (String) getParameter(messageContext, EmailConstants.SUBJECT);
+        String content = (String) getParameter(messageContext, EmailConstants.CONTENT);
+        String attachments = (String) getParameter(messageContext, EmailConstants.ATTACHMENTS);
+        String contentType = (String) getParameter(messageContext, EmailConstants.CONTENT_TYPE);
+        String encoding = (String) getParameter(messageContext, EmailConstants.ENCODING);
+        String contentTransferEncoding = (String) getParameter(messageContext, EmailConstants.CONTENT_TRANSFER_ENCODING);
+
+        if (StringUtils.isEmpty(to) && StringUtils.isEmpty(cc) && StringUtils.isEmpty(bcc)) {
+            throw new InvalidConfigurationException("Error occurred while sending the email. " +
+                    "Recipients are not provided.");
+        } else {
+            try {
+                //TODO: Set headers from transport properties
+                MimeMessage message = MessageBuilder.newMessage(session.getSession())
+                        .to(to)
+                        .fromAddresses(from)
+                        .cc(cc)
+                        .bcc(bcc)
+                        .replyTo(replyTo)
+                        .withSubject(subject)
+                        .withBody(content, contentType, encoding, contentTransferEncoding)
+                        .withAttachments(attachments)
+                        .build();
+                Transport.send(message);
+                log.debug("Email was sent successfully...");
```

Comment:
Shall we add some unique field of message to identify which message was sent successfully? Otherwise this log does not serve the purpose.

Reply:
Added subject and 'to' recipient. We do not have any unique identifier for the message assigned at this point